cluster/cluster.go

```diff
@@ -150,11 +160,49 @@ func ParseConfig(clusterFile string) (*v3.RancherKubernetesEngineConfig, error)
 	if err := yaml.Unmarshal([]byte(clusterFile), &rkeConfig); err != nil {
 		return nil, err
 	}
+	// the customConfig is mapped to a k8s type, which doesn't unmarshal well because it has a
+	// nested struct and no yaml tags. Therefor, we have to re-parse it again and assign it correctly.
+	// this only affects rke cli. Since rkeConfig is passed from rancher directly in the rancher use case.
+	parseCustomEncryptionConfig(clusterFile, &rkeConfig)
 	return &rkeConfig, nil
 }
 
-func InitClusterObject(ctx context.Context, rkeConfig *v3.RancherKubernetesEngineConfig, flags ExternalFlags) (*Cluster, error) {
+func parseCustomEncryptionConfig(clusterFile string, rkeConfig *v3.RancherKubernetesEngineConfig) error {
```

Comment:
This method belongs to utils. Lets put it to the go file where you do yaml manipulations

Reply:
It's encryption specific, so I moved it to encryption.go